internal/http/fetch_grant_token_test.go

```diff
@@ -0,0 +1,156 @@
+			authToken: "test-token",
+			serverFn: func(w http.ResponseWriter, _ *http.Request) {
+				w.Header().Set("Content-Type", "application/json")
+				w.WriteHeader(http.StatusOK)
+				if err := json.NewEncoder(w).Encode(map[string]interface{}{
+					"data": map[string]string{
+						"token": "test-grant-token",
+					},
+				}); err != nil {
+					t.Errorf("Failed to encode response: %v", err)
+				}
+			},
+		},
+		{
+			name:      "invalid response json",
+			authToken: "test-token",
+			serverFn: func(w http.ResponseWriter, _ *http.Request) {
+				w.Header().Set("Content-Type", "application/json")
+				w.WriteHeader(http.StatusOK)
+				if _, err := w.Write([]byte("invalid json")); err != nil {
+					t.Errorf("Failed to write response: %v", err)
+				}
+			},
+			wantErr:       true,
+			errorContains: "failed to decode grant token response",
+		},
+		{
+			name:      "server error",
+			authToken: "test-token",
+			serverFn: func(w http.ResponseWriter, _ *http.Request) {
+				w.WriteHeader(http.StatusInternalServerError)
+			},
+			wantErr:       true,
+			errorContains: "status code 500",
+		},
+		{
+			name:      "verify request body",
+			authToken: "test-auth-token",
+			serverFn: func(w http.ResponseWriter, r *http.Request) {
+				var body struct {
+					Token string `json:"token"`
+				}
+				if err := json.NewDecoder(r.Body).Decode(&body); err != nil {
+					t.Errorf("Failed to decode request body: %v", err)
+				}
+				if body.Token != "test-auth-token" {
+					t.Errorf("Expected auth token 'test-auth-token', got %q", body.Token)
+				}
+				if r.Header.Get("Content-Type") != "application/json" {
+					t.Errorf("Expected Content-Type 'application/json', got %q", r.Header.Get("Content-Type"))
+				}
+
+				w.Header().Set("Content-Type", "application/json")
+				w.WriteHeader(http.StatusOK)
+				if err := json.NewEncoder(w).Encode(map[string]interface{}{
+					"data": map[string]string{
+						"token": "test-grant-token",
+					},
+				}); err != nil {
+					t.Errorf("Failed to encode response: %v", err)
+				}
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			server := httptest.NewServer(http.HandlerFunc(tt.serverFn))
+			defer server.Close()
+
+			token, err := FetchGrantToken(server.URL, tt.authToken)
+			hasErr := err != nil
+
+			if hasErr != tt.wantErr {
+				t.Errorf("FetchGrantToken() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			if hasErr && tt.wantErr && !strings.Contains(err.Error(), tt.errorContains) {
+				t.Errorf("Expected error containing %q, got %v", tt.errorContains, err)
+			}
+
+			if !tt.wantErr && token == "" {
+				t.Error("Expected non-empty token for successful request")
+			}
+		})
+	}
+}
+
+func TestFetchGrantTokenInvalidURL(t *testing.T) {
+	_, err := FetchGrantToken("not-a-valid-url", "test-token")
+	if err == nil {
+		t.Error("Expected error for invalid URL, got nil")
+	}
+}
+
+func TestFetchGrantTokenRetry(t *testing.T) {
```

Comment:
Would it be possible to test also failing connection to the server? Feel free to ignore if gets to cumbersome

Reply:
[84e4798](https://github.com/n8n-io/task-runner-launcher/pull/38/commits/84e4798a0c9e37a9470aeecb3b451ec25f0e7b26)